components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -938,13 +939,24 @@ public static String resolveEventName(String notificationChannel) {
     /**
      * Validate email username.
      *
-     * @param username Tenant aware username of the user.
-     * @throws IdentityRecoveryClientException If username is not an email when email username is enabled.
+     * @param user User .
+     * @throws IdentityRecoveryClientException If username is not an email for tenanted users when email username
+     * is enabled.
      */
-    public static void validateEmailUsername(String username) throws IdentityRecoveryClientException {
+    public static void validateEmailUsername(User user) throws IdentityRecoveryClientException {
```

Comment:
we should not change the signatures of public methods. since this is a util method, it can be used by extensions as well. Any reasons for changing the public method. 
Ideally, we should have deprecated the previous method and introduced a new method for user object

Reply:
To fix https://github.com/wso2/product-is/issues/13498, we need to update the logic of this method, To do that we need some additional information like tenant domain. That's the reason why I used the User object instead of String username. I'll add this as a separate method as you suggested and deprecate the previous method.

Reply:
Did we remove this in wum and u2 prs too? If so, please fix them too